lib/bolt/module_installer/installer.rb

```diff
@@ -28,14 +29,17 @@ def install(path, moduledir)
           moduledir:  moduledir.to_s
         }
 
+        @outputter.start_spin
```

Comment:
I think it'd be helpful to have the outputter spin for _all_ of the `module add|install` command, not just the install step. Resolving modules in particular can take a while, at times even longer than installing them.

Reply:
Hm, yeah. I was initially thinking it'd be best to have the spinner spin closest to the long running operation themselves, so that any codepaths that got to that operation would use the spinner and there was less risk of having two spinners at once. That means passing the outputter around a lot though, and while I'm not sure anyone else is going to be using the module_installer and project migrator it doesn't quite feel right to *require* the outputter there. Then again, it does feel much less readable and discoverable when the calling methods are wrapped in spins and not the "end" of the codepath itself....I don't know I'm still on the fence.

Reply:
I think having the spinners all in the CLI wrapping the entire operation makes things more discoverable, instead of needing to hunt down where the spinner is. It also seems a bit cleaner - as you noted with the outputter needing to be passed around a lot, not having to do that would keep function signatures simpler.

I think either approach is fine and don't feel super strongly one way or the other. 🤷

Reply:
I don't think we can wrap in the CLI for things like the migrator because it will write the spinner to the prompt when we prompt for user input. But the more I think about it the more I like keeping the outputter out of the installer and resolver classes, so I like the approach I have here.